pkg/realizer/component.go

```diff
@@ -31,49 +32,64 @@ import (
 	"github.com/vmware-tanzu/cartographer/pkg/templates"
 )
 
-//go:generate go run -modfile ../../../hack/tools/go.mod github.com/maxbrunsfeld/counterfeiter/v6 -generate
+//go:generate go run -modfile ../../hack/tools/go.mod github.com/maxbrunsfeld/counterfeiter/v6 -generate
+
+type OwnerResource struct {
+	TemplateRef     v1alpha1.TemplateReference
+	TemplateOptions []v1alpha1.TemplateOption
+	Params          []v1alpha1.BlueprintParam
+	Name            string
+	Sources         []v1alpha1.ResourceReference
+	Images          []v1alpha1.ResourceReference
+	Configs         []v1alpha1.ResourceReference
+	Deployment      *v1alpha1.DeploymentReference
+}
 
 //counterfeiter:generate . ResourceRealizer
 type ResourceRealizer interface {
-	Do(ctx context.Context, resource *v1alpha1.SupplyChainResource, supplyChainName string, outputs Outputs) (templates.Template, *unstructured.Unstructured, *templates.Output, error)
+	Do(ctx context.Context, resource OwnerResource, blueprintName string, outputs Outputs) (templates.Template, *unstructured.Unstructured, *templates.Output, error)
 }
 
 type resourceRealizer struct {
-	workload          *v1alpha1.Workload
-	systemRepo        repository.Repository
-	workloadRepo      repository.Repository
-	supplyChainParams []v1alpha1.BlueprintParam
+	owner           client.Object
+	ownerParams     []v1alpha1.OwnerParam
+	systemRepo      repository.Repository
+	ownerRepo       repository.Repository
+	blueprintParams []v1alpha1.BlueprintParam
+	resourceLabeler ResourceLabeler
 }
 
-type ResourceRealizerBuilder func(secret *corev1.Secret, workload *v1alpha1.Workload, systemRepo repository.Repository, supplyChainParams []v1alpha1.BlueprintParam) (ResourceRealizer, error)
+type ResourceRealizerBuilder func(secret *corev1.Secret, owner client.Object, ownerParams []v1alpha1.OwnerParam, systemRepo repository.Repository, blueprintParams []v1alpha1.BlueprintParam, resourceLabeler ResourceLabeler) (ResourceRealizer, error)
```

Comment:
I noticed this is getting quite wide 😅 

wdyt of this style?

```diff
diff --git a/pkg/realizer/component.go b/pkg/realizer/component.go
index c7ecf8a..eaad63b 100644
--- a/pkg/realizer/component.go
+++ b/pkg/realizer/component.go
@@ -59,11 +59,30 @@ type resourceRealizer struct {
        resourceLabeler ResourceLabeler
 }

-type ResourceRealizerBuilder func(secret *corev1.Secret, owner client.Object, ownerParams []v1alpha1.OwnerParam, systemRepo repository.Repository, blueprintParams []v1alpha1.BlueprintParam, resourceLabeler ResourceLabeler) (ResourceRealizer, error)
+type ResourceRealizerBuilder func(
+       secret *corev1.Secret,
+       owner client.Object,
+       ownerParams []v1alpha1.OwnerParam,
+       systemRepo repository.Repository,
+       blueprintParams []v1alpha1.BlueprintParam,
+       resourceLabeler ResourceLabeler,
+) (ResourceRealizer, error)

 //counterfeiter:generate sigs.k8s.io/controller-runtime/pkg/client.Client
-func NewResourceRealizerBuilder(repositoryBuilder repository.RepositoryBuilder, clientBuilder realizerclient.ClientBuilder, cache repository.RepoCache) ResourceRealizerBuilder {
-       return func(secret *corev1.Secret, owner client.Object, ownerParams []v1alpha1.OwnerParam, systemRepo repository.Repository, supplyChainParams []v1alpha1.BlueprintParam, resourceLabeler ResourceLabeler) (ResourceRealizer, error) {
+func NewResourceRealizerBuilder(
+       repositoryBuilder repository.RepositoryBuilder,
+       clientBuilder realizerclient.ClientBuilder,
+       cache repository.RepoCache,
+) ResourceRealizerBuilder {
+       return func(
+               secret *corev1.Secret,
+               owner client.Object,
+               ownerParams []v1alpha1.OwnerParam,
+               systemRepo repository.Repository,
+               supplyChainParams []v1alpha1.BlueprintParam,
+               resourceLabeler ResourceLabeler,
+       ) (ResourceRealizer, error) {
+
                ownerClient, _, err := clientBuilder(secret, false)
                if err != nil {
                        return nil, fmt.Errorf("can't build client: %w", err)
```

Reply:
no strong opinions - I'm more used to multiline, but I think most core contributors are not 😁  thx!